db/model/subject.js

```diff
@@ -344,8 +344,24 @@ module.exports = function subject(seq, dataTypes) {
 
         if (inst.getDataValue('isPublished')) {
           if (inst.previous('isPublished')) {
-            common.publishChange(inst, eventName.upd, changedKeys,
+            /*
+             * If tags were updated, send a "delete" event followed by an "add"
+             * event so that perspectives using subject tag filters will get
+             * the right realtime events. If subject tags were not updated,
+             * just send the usual "update" event.
+             *
+             * TODO : Right now don't have the ability to mock the socket.io
+             * test for this.
+             */
+            if (inst.changed('tags')) {
+              common.publishChange(inst, eventName.del, ['tags'],
+                ignoreAttributes);
+              common.publishChange(inst, eventName.add, ['tags'],
+                ignoreAttributes);
+            } else {
+              common.publishChange(inst, eventName.upd, changedKeys,
               ignoreAttributes);
+            }
           } else {
```

Comment:
The changes you did in the utils will take care of everything.  Can you test without these changes to make sure?